src/app_client.rs

```diff
@@ -373,20 +447,114 @@ pub async fn run(
+				188, 122, 148, 8, 106, 162, 124, 253, 119, 219, 23, 58, 172, 0, 128, 56, 149, 136,
+				107, 138, 79, 73, 232, 92, 104, 244, 105, 213, 112, 240, 237, 153, 39, 80, 191,
+				149, 50, 155, 185, 14, 245, 107, 69, 171, 205, 0, 159, 237, 239, 13, 156, 189, 214,
+				28, 5, 161, 129, 212, 1, 56, 0, 4, 29, 1, 33, 3, 48, 54, 52, 50, 101, 53, 100, 48,
+				52, 98, 54, 0, 54, 50, 53, 57, 102, 54, 54, 53, 102, 100, 49, 53, 51, 97, 97, 49,
+				54, 100, 102, 55, 52, 48, 102, 50, 53, 51, 55, 50, 55, 56, 102, 0, 97, 49, 57, 49,
+				101, 101, 57, 54, 48, 52, 56, 98, 102, 56, 57, 57, 55, 51, 52, 57, 97, 48, 49, 55,
+				53, 56, 101, 52, 98, 55, 97, 0, 50, 100, 53, 57, 102, 53, 52, 53, 51, 56, 57, 56,
+				101, 98, 98, 49, 100, 50, 51, 98, 102, 52, 53, 57, 101, 54, 54, 55, 97, 54, 51, 0,
+				52, 98, 49, 54, 99, 102, 52, 50, 50, 102, 99, 57, 51, 53, 51, 100, 52, 98, 56, 98,
+				98, 54, 48, 56, 98, 53, 57, 50, 48, 101, 52, 0, 53, 55, 51, 102, 51, 53, 102, 48,
+				55, 48, 55, 100, 51, 50, 56, 97, 102, 97, 52, 56, 50, 49, 102, 99, 101, 102, 49,
+				54, 52, 57, 102, 0, 57, 100, 53, 50, 101, 55, 98, 53, 55, 50, 53, 101, 51, 48, 57,
+				53, 100, 56, 101, 101, 101, 97, 53, 100, 54, 99, 50, 53, 51, 56, 48, 0, 100, 52,
+				101, 128, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+				0, 0, 0, 0, 52, 96, 128, 190, 131, 244, 138, 209, 116, 140, 74, 211, 57, 171, 220,
+				184, 3, 54, 142, 253, 209, 246, 86, 137, 97, 159, 248, 194, 8, 117, 93, 0, 132,
+				238, 252, 248, 55, 182, 28, 71, 155, 51, 50, 5, 155, 200, 232, 155, 73, 10, 157,
+				80, 43, 174, 202, 237, 68, 132, 51, 212, 225, 97, 113, 0, 0, 167, 28, 187, 26, 3,
+				135, 89, 142, 80, 157, 159, 202, 181, 17, 2, 47, 67, 123, 12, 175, 19, 89, 19, 21,
+				195, 241, 187, 240, 79, 24, 0, 157, 131, 240, 20, 128, 98, 16, 218, 110, 225, 210,
+				248, 12, 240, 249, 192, 143, 29, 19, 43, 224, 66, 118, 144, 21, 246, 23, 79, 210,
+				178, 76, 0,
+			]),
+			None,
+		]
+		.to_vec();
+
+		let block = BlockVerified {
+			header_hash: hex!("5bc959e1d05c68f7e1b5bc3a83cfba4efe636ce7f86102c30bcd6a2794e75afe")
+				.into(),
+			block_num: 288,
+			dimensions,
+			lookup: AppDataIndex {
+				size: 12,
+				index: [(1, 1)].to_vec(),
+			},
+			commitments: [
+				[
+					165, 227, 207, 130, 59, 77, 78, 242, 184, 232, 114, 218, 145, 167, 149, 53, 89,
+					7, 230, 49, 85, 113, 218, 116, 43, 195, 144, 203, 149, 114, 106, 89, 73, 164,
+					17, 163, 3, 145, 173, 6, 119, 222, 17, 60, 251, 215, 40, 192,
+				],
+				[
+					165, 227, 207, 130, 59, 77, 78, 242, 184, 232, 114, 218, 145, 167, 149, 53, 89,
+					7, 230, 49, 85, 113, 218, 116, 43, 195, 144, 203, 149, 114, 106, 89, 73, 164,
+					17, 163, 3, 145, 173, 6, 119, 222, 17, 60, 251, 215, 40, 192,
+				],
+			]
+			.to_vec(),
+		};
+		mock_client
+			.expect_fetch_rows_from_dht()
+			.returning(move |_, _, _| {
+				let dht_rows_clone = dht_rows.clone();
+				Box::pin(async move { dht_rows_clone })
+			});
+		if cfg.disable_rpc {
```

Comment:
Each test run will have same disable_rpc value, we should perform two process_block runs with different configurations and expectations, and that way we will be sure that process block behaves correctly in both cases.